text/primitive-projections.md

```diff
@@ -0,0 +1,294 @@
+   - Advantages: the natural expectation
+   - Drawbacks: maybe a bit ad hoc, though controlling reduction with properties of the arguments is not new (already done for `simpl`)
+
+Related: [#14084](https://github.com/coq/coq/pull/14084) (`match goal` involving primitive projections) and [#11366](https://github.com/coq/coq/issues/11366) (which requires to be able to declare partially applied projections as coercions).
+
+## Issue 2: the relative status of "folded" vs "unfolded" `Proj` node
+
+*Design choice 1*: The "folded" vs "unfolded" boolean is dropped. Unfolding a `Proj` node is a no-op. In scenario 2.1 of issue 1, unfolding a `Const` denoting a projection reveals a `Proj` node. In scenario 2.2, it is a no-op.
+
+*Design choice 2*: Idem but unfolding a primitive projection applied to the constructor of the record performs the iota-reduction (like in the positive case) instead of a being a no-op.
+
+Related: [#5250](https://github.com/coq/coq/issues/5250), [#5698](https://github.com/coq/coq/issues/5698), [#5699](https://github.com/coq/coq/issues/5699) (effect of unfolding tactics).
+
+See associated [Zulip discussion](https://coq.zulipchat.com/#narrow/stream/237656-Coq-devs.20.26.20plugin.20devs/topic/Primitive.20Projection.20mode).
+
+Note: effect of various reduction commands on the diverse representations of projections:
+
+|                                | non-primitive | prim Const        | prim folded Proj | prim unfolded Proj | expected
+| ------------------------------ | ------------- | ----------------- | ---------------- | ------------------ | --------
+| unfold on non-`Build`          | to `match`    | to unfolded Proj  | to unfolded Proj | nop                | error (2) or nop (1)
+| unfold at on non-`Build`       | to `match`    | to unfolded Proj  | to unfolded Proj | error              | error or nop?
+| unfold on `Build`              | contract      | contract          | contract         | contract           | contract
+| unfold at on `Build`           | contract      | error (bug?)      | error (bug?)     | contract           | contract
+| simpl on arg non-red `Build`   | nop           | nop               | nop              | nop                | nop
+| simpl on arg red `Build`       | contract      | contract          | contract         | contract           | contract
+| simpl never                    | nop           | nop               | nop              | nop                | nop
+| simpl nomatch non-red `Build`  | nop           | nop               | nop              | nop                | nop
+| delta                          | to `match`    | to expanded unfolded Proj | to unfolded Proj | nop        | nop
+| iota on `Build`                | contract      | nop               | nop              | nop (bug?)         | contract
```

Comment:
At least to go from folded to unfolded, we need `beta delta`, not `delta`; I expect `iota` would then work.

https://coq.zulipchat.com/#narrow/stream/237977-Coq-users/topic/Unfolding.20prim.20projections.20requires.20.60beta.60.20.E2.80.94.20intended.3F

Reply:
There are several errors indeed.
- I failed to see that `delta` was not working alone on "folded"
- The assumption I made on preparing tests on "unfolded" was wrong (due to using `delta` instead of `beta delta`) and thus `iota` but also `simpl never` on unfolded work
- The `unfold at` tests were otherwise mixed up.
- I should have distinguished "match" and "expanded match"

Reply:
We might want better inspection options as a first step…

Reply:
In the last version, to inspect folded vs unfolded, I used a `match goal` which seems correctly discriminating. If you have time, that'll be good to double-check the output of the tests.